src/mesido/workflows/io/write_output.py

```diff
@@ -296,7 +297,16 @@ def _write_html_output(self, template_name="mpc_buffer_sizing_output"):
                 )
             )
 
-    def _write_updated_esdl(self, energy_system, optimizer_sim: bool = False):
+    def _write_updated_esdl(
+        self,
+        energy_system,
+        network_type,
+        optimizer_sim: bool = False,
+    ):
+        if network_type != NetworkSettings.NETWORK_TYPE_HEAT:
+            logger.error(f"_write_updated_esdl does not support network type: {network_type}")
+            sys.exit(1)
```

Comment:
This is exactly the point of this exit. To force one to update the _write_updated_esdl. Currently the only assets be updated (placement) are heat_pipe, heat_source, ates, heat_buffer. So no gas and electricity assets are updated

Reply:
I would prefer not to have the network_type used for this check, but check which types of components are in the ESDL, and provide logger warning that types x,y,z are not updated in the ESDL. Because I for see sometime soon, we'll use HPs with small electricity connection (just one cable and ELproducer), I don't want it to break the ESDL update, but only provide a warning that the ELproducer and cable are not updated.

Reply:
Ah ok. If you only want to size the heat pump, it would make sense. I will update the code accordingly

Reply:
The change has been made